### PR TITLE
Add NamedParam support for gradle-2.5 with backport to 2.4

### PR DESCRIPTION
**Pull request by leonard84**
This change is [Reviewable](https://reviewable.io/reviews/spockframework/spock/921)

### CODE REVIEW

**Comment by leonard84** [comment]
@marcphilipp how and where would you add this mention?

**Comment by marcphilipp** [comment]
In the Javadoc of the classes so it's at least clear to contributors and maintainers. 😉

File: internal-backport/src/main/java/groovy/transform/NamedParam.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *  under the License.
+ */
+package groovy.transform;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+// Backport so that we can add it for 2.5 but can still build with 2.4 as well
```
Comment (leonard84) [inline internal-backport/src/main/java/groovy/transform/NamedParam.java]: So this is not sufficient ;)

Comment (marcphilipp) [reply internal-backport/src/main/java/groovy/transform/NamedParam.java]: To me that sentence does not clearly express that it won't work with 2.4. Of course, that's logical when you think about it. But a "backport" is usually a feature ported from a newer to an older version, i.e. something that actually works. Might be just me, though. I'll leave it up to you to decide. 🙂

**Pull request merged**